### DISCUSSION TITLE
Migrate Historical Recurring Donations with Salesforce add on

### DISCUSSION THREAD

**Discussion by merguk** [impress-org/givewp#6871] [Developer Q&A]
Salesforce partner implementing GiveWP and recurring donations here.

We've turned on the integration and **new subscribers** are working, syncing to SF, along with Recurring Donation and monthly Opportunities.  Not seeing anything for existing subscribers though.  What's the expected behavior?

Is there any way to force Givewp to attempt the 'first' sync on the subscription, and the subsequent donations, for pre integration subscribers?

Or if that's not possible, what's the data it looks for, to sync new donations to?  For example, I imagine every month's donation will be attempted to sync, and if no Recurring Donation (subscription) record is found, it will fail.  So if we created these records manually, does GiveWP try to sync against a particular ID, or just any Open Recurring Donation against the relevant Contact?

Thanks

**Comment by jonwaldstein** [comment]
Hi @merguk

The Salesforce add-on only works for new subscriptions.  It _will_ attempt to find an existing contact by meta key `_give_donor_salesforce_contact_id` then search by name & email.  If none found will create new contact and store that meta key to the database for future requests.

All incoming renewals check if their subscription in GiveWP has the meta key `_give_subscription_salesforce_recurring_donation_id` - otherwise will create a new one in Salesforce and store the id in the database for future requests.  Then the renewal attempts to check if it exists in Salesforce based on installment pledged close date - otherwise will create a new one.

**Reply by merguk** [reply]
Awesome, thanks @jonwaldstein, exactly what I was looking for.  Are these ids accessible for us to populate in GiveWP manually?

```
_give_donor_salesforce_contact_id
_give_subscription_salesforce_recurring_donation_id
```

I'm thinking of manually creating them in SF, linking them with above fields in GiveWP, and hopefully monthly donations should start flowing into SF, from the pre integration subscribers.

Cheers
Mergen

**Reply by jonwaldstein** [reply]
@merguk yes it's technically possible, you would just need to manually pull these id's from salesforce and update the GiveWP meta accordingly.  

These are all the meta keys we use:
```
_give_donor_salesforce_contact_id - give_donormeta
_give_subscription_salesforce_recurring_donation_id - give_subscriptionmeta
_give_donation_salesforce_opportunity_id - give_donationmeta (stored for one-time donations and subscription renewals)
```
